src/components/Button/Button.tsx

```diff
@@ -1,28 +1,28 @@
-import { CSSProperties, ReactChild, ReactElement } from "react";
+import { CSSProperties, ReactElement } from "react";
 import styled, { css } from "styled-components";
 
-export interface Props {
-  title: string;
+/**
+ * A group of properties for `<Root />`. If a value can be used in
+ * anywhere else, set it on `Props` instead.
+ */
+interface RootBodyProps {
   textLeft?: boolean;
-  icon?: ReactElement;
   onClick?: () => void;
   className?: string;
   style?: CSSProperties;
 }
 
-export default function Button({
-  title,
-  textLeft,
-  icon,
-  onClick,
-  ...props
-}: Props) {
+export interface Props extends RootBodyProps {
+  title: string;
+  icon?: ReactElement;
+}
+
+export default function Button({ title, icon, ...rootBodyProps }: Props) {
```

Comment:
I came across this file to add `data-testid` property for a test but then figured that I could tweak the props of this file a little. I coupled all props that are passed for `Root` for readability. Lemme know what you think, I'm totally fine reverting this change as it's just my preference here😄